lumicks/pylake/force_calibration/power_spectrum.py

```diff
@@ -19,30 +20,52 @@ class PowerSpectrum:
         The total duration of the original data. [seconds]
     """
 
-    def __init__(self, data, sample_rate, unit="V"):
+    def __init__(self, data, sample_rate, unit="V", window_seconds=None):
         """Power spectrum
 
         Parameters
         ----------
         data : numpy.ndarray
             Data from which to calculate a power spectrum.
         sample_rate : float
-            Sampling rate at which this data was acquired.
+            Sampling rate at which this data was acquired [Hz].
         unit : str
             Units the data is in (default: V).
+        window_seconds : float
+            Window duration [seconds]. When specified the data is divided into blocks of length
+            window_seconds. Power spectra are computed for each block after which they are
+            averaged. If omitted, no windowing is used.
         """
+
+        def squared_fft(d):
+            return np.square(np.abs(np.fft.rfft(d)))
+
         self.unit = unit
         data = data - np.mean(data)
 
-        # Calculate power spectrum.
-        fft = np.fft.rfft(data)
-        self.frequency = np.fft.rfftfreq(data.size, 1.0 / sample_rate)
-        self.power = (2.0 / sample_rate) * np.square(np.abs(fft)) / data.size
+        # Calculate power spectrum for slices of data.
+        num_points_per_window = (
+            int(np.round(window_seconds * sample_rate)) if window_seconds else len(data)
+        )
```

Comment:
should we check for negative values in `window_seconds`?

Reply:
Added!